marathon_sync/main.py

```diff
@@ -0,0 +1,135 @@
+import json
+import treq
+
+from twisted.python import log
+from twisted.web import client
+# Twisted's fault HTTP11 client factory is way too verbose
+client._HTTP11ClientFactory.noisy = False
+from twisted.internet.defer import (
+    gatherResults, inlineCallbacks, returnValue)
+
+
+def normalise_group_id(group_id):
+    """
+    Group IDs can be specified with or without leading and trailing '/'s. We
+    want a leading '/' but not a trailing one.
+    """
+    if not group_id.startswith('/'):
+        group_id = '/' + group_id
+
+    if group_id.endswith('/') and group_id != '/':
+        group_id = group_id[:-1]
+
+    return group_id
```

Comment:
Would it not be simpler to just strip all leading and trailing slashes and then prepend a leading slash?

Something like `return '/' + group_id.strip('/')` perhaps.